### PR TITLE
PHPC-428: Remove support for WriteConcern fsync option

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-428

### CODE REVIEW

File: src/MongoDB/WriteConcern.c [hunk 1]
```diff
@@ -66,7 +64,7 @@ PHP_METHOD(WriteConcern, __construct)
 	zend_replace_error_handling(EH_THROW, phongo_exception_from_phongo_domain(PHONGO_ERROR_INVALID_ARGUMENT), &error_handling TSRMLS_CC);
 	intern = (php_phongo_writeconcern_t *)zend_object_store_get_object(getThis() TSRMLS_CC);
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "z|lb!b!", &w, &wtimeout, &journal, &journal_is_null, &fsync, &fsync_is_null) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "z|lb!b!", &w, &wtimeout, &journal, &journal_is_null) == FAILURE) {
```
Comment (derickr) [inline src/MongoDB/WriteConcern.c]: You missed removing the "b!" from the zpp call.

File: php_phongo.c [hunk 2]
```diff
@@ -1432,11 +1432,7 @@ void php_phongo_write_concern_to_zval(zval *retval, const mongoc_write_concern_t
 
 	add_assoc_bool_ex(retval, ZEND_STRS("wmajority"), mongoc_write_concern_get_wmajority(write_concern));
 	add_assoc_long_ex(retval, ZEND_STRS("wtimeout"), mongoc_write_concern_get_wtimeout(write_concern));
-	if (write_concern->fsync_ != MONGOC_WRITE_CONCERN_FSYNC_DEFAULT) {
-		add_assoc_bool_ex(retval, ZEND_STRS("fsync"), mongoc_write_concern_get_fsync(write_concern));
-	} else {
-		add_assoc_null_ex(retval, ZEND_STRS("fsync"));
-	}
+
```
Comment (bjori) [inline php_phongo.c]: You should update the array_init_size

File: src/MongoDB/WriteConcern.c (hunk 1 above)
Comment (bjori) [inline src/MongoDB/WriteConcern.c]: This syntax is not PHP5.4 compatible

File: php_phongo.c (hunk 2 above)
Comment (jmikola) [inline php_phongo.c]: Good catch :+1:

File: src/MongoDB/WriteConcern.c (hunk 1 above)
Comment (jmikola) [inline src/MongoDB/WriteConcern.c]: You would be correct! https://travis-ci.org/mongodb-labs/mongo-php-driver-prototype/jobs/81668599#L1291

I'll change this to `z` and manually check for null values.

**Pull request merged**